sno/upgrade/upgrade_v0.py

```diff
@@ -0,0 +1,265 @@
+import functools
+import json
+import re
+import subprocess
+from collections import deque
+from datetime import datetime
+from pathlib import Path
+
+import click
+import pygit2
+
+from sno import checkout, context, gpkg_adapter
+from sno.exceptions import InvalidOperation
+from sno.fast_import import fast_import_tables
+from sno.geometry import normalise_gpkg_geom
+from sno.repository_version import get_repo_version, write_repo_version_config
+from sno.schema import Schema
+from sno.structure import DatasetStructure
+
+
+def upgrade_v0(ctx, source, dest):
+    """
+    Upgrade a v0.0/v0.1 Sno repository to Sno v0.2
+    """
+    source = Path(source)
+    dest = Path(dest)
+
+    if dest.exists():
+        raise click.BadParameter(f"'{dest}': already exists", param_hint="DEST")
+
+    source_repo = pygit2.Repository(str(source))
+    if not source_repo or not source_repo.is_bare:
+        raise click.BadParameter(
+            f"'{source}': not an existing repository", param_hint="SOURCE"
+        )
+
+    version = get_repo_version(source_repo)
+    if version != 0:
+        raise InvalidOperation(
+            "source repo is not at sno version 0.2 - (dataset version 1)"
```

Comment:
wrong version in message

Reply:
Fixed - no longer need this check anyway now that this part of the code is the same for both versions.